### ISSUE TITLE
ab-option-021 should cause err:XD0036

### ISSUE THREAD

**Issue by ndw** [xproc/3.0-test-suite#22]
The description for `ab-option-021` is:
> Checks not required option with default value is type casted.

The relevant option is:

```xml
<p:option name="option" required="false" select="'false'" as="xs:boolean"/>
```

I think this test should raise err:XD0036. I don’t think we should ever cast strings to other values. If the author wants false there, the simple thing to do is say `select="false()"`.

I think we have a special case for values that are of type `xs:untypedAtomic`. That makes it easier to pass options in from outside the pipeline.

Or perhaps I misunderstand the test.

**Comment by xml-project** [comment]
I think the motivation is [this](http://spec.xproc.org/master/head/xproc/#p.with-option):

> If an atomic type, or sequence of atomic types, is specified, the value provided for the option will be atomized according to the standard XPath rules.

If you take "standard XPath rules" to mean [XPath's function conversion rules](https://www.w3.org/TR/xpath-31/#id-function-conversion-rules), I think casting has to be done.

But obviously I misunderstood "standard XPath rules".

**Comment by ndw** [comment]
The value `select="'false'"` is a string, it will be atomized, unsurprisingly I hope, as a string.

The value `select="'false' cast as xs:boolean" is a boolean and will be atomized as a boolean.

The value `select="3"` is a number and will be atomized as a, uh, `xs:decimal` by default I think.

Unless I misunderstand the rules.

**Comment by xml-project** [comment]
Yes, you are absolutely right. I am was not arguing for the test, I was explaining why I got it wrong.

**Comment by ndw** [comment]
Ah. Ok. So I suggest a PR to have that test raise the appropriate error code, unless there's already a different test for that.

**Comment by xml-project** [comment]
Working on it. Found another test, where we should move to xs0089.

**Comment by xml-project** [comment]
Fixed with pr #24
